scrapy/core/downloader/handlers/http11.py

```diff
@@ -445,6 +445,14 @@ def __init__(self, finished, txresponse, request, maxsize, warnsize, fail_on_dat
         self._fail_on_dataloss_warned = False
         self._reached_warnsize = False
         self._bytes_received = 0
+        self._certificate = None
+
+    def connectionMade(self):
+        try:
+            assert self._certificate is None
```

Comment:
We should not use assert statements in production code, outside tests, where we know the assertion may fail. Assert statements [are removed when optimizations are enabled](https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement).

If you believe the assert can raise an `AssertionError` under some circumstances, better use an `if` statement instead.

Reply:
Updated, thanks :raised_hands: